push/parser.go

```diff
@@ -46,11 +46,12 @@ const (
 	occupancyPrefix = "[?occupancy=metrics.publishers]"
 )
 
-const (
-	NotCompressed = iota
-	Gzip
-	Zlib
-)
+func getCompressType(number *int) *int {
```

Comment:
I will continue using enum and assign it and default to zero or nonValue -1 if you like